modules/compiler/utils/source/handle_barriers_pass.cpp

```diff
@@ -1143,7 +1143,36 @@ Function *compiler::utils::HandleBarriersPass::makeWrapperFunction(
         auto *MemBarrier = BI.getOrDeclareMuxBuiltin(eMuxBuiltinMemBarrier, M);
         assert(MemBarrier);
         Value *Ops[2] = {CI->getOperand(1), CI->getOperand(2)};
-        auto *Call = B.CreateCall(MemBarrier, Ops);
+
+        auto *const Call = B.CreateCall(MemBarrier, Ops);
+
+        // Patch up any operands that were non-constants by fetching them from
+        // the barrier struct. We do this after creating the call because we
+        // need an instruction to function as an insert point.
+        if (!isa<Constant>(Ops[0]) || !isa<Constant>(Ops[1])) {
+          dbgs() << *Call << "\n";
+          // We expect these values to be uniform so it should be safe to get
+          // from the barrier struct at index zero. Barriers are convergent, so
+          // there should be no chance that the value does not exist.
+          auto *const zero =
+              Constant::getNullValue(Type::getIntNTy(context, 8 * sizeTyBytes));
```

Comment:
hmm? it's a variable local to this function.

Reply:
Oh sorry I meant we could use `llvm::IntegerType *getSizeType(const llvm::Module &m)`, to avoid the internal details of `Type::getIntNTy(context, 8 * sizeTyBytes)`

Reply:
ah yes.. i don't know why we've always done it this way.. it's all through this entire pass though.

Reply:
decided not to do anything about it now